docs/notebooks/ask_tell_optimization.pct.py

```diff
@@ -0,0 +1,166 @@
+# %% [markdown]
+# # Ask-Tell Optimization Interface
+
+# %% [markdown]
+# In this notebook we will illustrate the use of Ask-Tell interface in Trieste. It might be useful in the cases when you want to have greater control of the optimization loop, or if letting Trieste manage this loop is impossible.
+#
+# First, some code to setup the problem we will be using throughout the notebook. If you would like more details about this problem setup, please refer to [introductory EI notebook](expected_improvement.ipynb).
+
+# %%
+import numpy as np
+import tensorflow as tf
+import matplotlib.pyplot as plt
+import gpflow
+
+from trieste.ask_tell_optimization import AskTellOptimizer
+from trieste.bayesian_optimizer import Record
+from trieste.data import Dataset
+from trieste.models import create_model
+from trieste.models.gpflow import GPflowModelConfig
+from trieste.objectives import scaled_branin, SCALED_BRANIN_MINIMUM
+from trieste.objectives.utils import mk_observer
+from trieste.observer import OBJECTIVE
+from trieste.space import Box
+
+from util.plotting import plot_regret
+
+np.random.seed(1234)
+tf.random.set_seed(1234)
+
+
+search_space = Box([0, 0], [1, 1])
+n_steps = 5
+
+def build_model(data, kernel_func=None):
+    """kernel_func should be a function that takes variance as a single input parameter"""
+    variance = tf.math.reduce_variance(data.observations)
+    if kernel_func is None:
+        kernel = gpflow.kernels.Matern52(variance=variance)
+    else:
+        kernel = kernel_func(variance)
+    gpr = gpflow.models.GPR(data.astuple(), kernel, noise_variance=1e-5)
+    gpflow.set_trainable(gpr.likelihood, False)
+
+    return GPflowModelConfig(**{
```

Comment:
you can simplify things here a bit by returning a `GaussianProcessRegression` object instead, then below you don't need to use `create_model` function

Reply:
fair enough, changed it